### ISSUE TITLE
"React does not recognise the `isFullWidth` prop on a DOM element" on Select component

### ISSUE THREAD

**Issue by mike-fam** [chakra-ui/chakra-ui#2865]
# 🐛 Bug report

When passing the isFullWidth prop to the Select component, the width of the component is not changed and this warning from React is logged to the terminal. 
```
Warning: React does not recognize the `isFullWidth` prop on a DOM element. If you intentionally want it to appear in the DOM as a custom attribute, spell it as lowercase `isfullwidth` instead. If you accidentally passed it from a parent component, remove it from the DOM element.
```
## 💥 Steps to reproduce
Add this component to a chakra project (I used the create-react-app template)
```react
<Select isFullWidth={false}>
    <option value="Hello">
        Hello
    </option>
    <option value="World">
        World
    </option>
</Select>
```

## 💻 Link to reproduction

[CodeSandbox](https://codesandbox.io/s/chakra-react-doesnt-recognise-c1v3m?file=/src/index.tsx)

## 🧐 Expected behavior

The Select component's width should be reduced to just be enough to contain the options.

## 🌍 System information

| Software         | Version(s) |
| ---------------- | ---------- |
| Chakra UI        |        1.0.3    |
| Browser          |      Chrome, Firefox      |
| Operating System |      MacOs      |

## 📝 Additional information

This worked about a month ago but doesn't work anymore. I believe the problem also exists for the `rightIcon` prop of the `MenuButton` component.

**Comment by artpumpkin** [comment]
I have the same issue with other props, like
> React does not recognize the `whileTap` prop on a DOM element. If you intentionally want it to appear in the DOM as a custom attribute, spell it as lowercase `whiletap` instead. If you accidentally passed it from a parent component, remove it from the DOM element.

other props: `whileHover`, `endColor`, `startColor`

I reckon it needs some sort of filtering.

**Comment by TimKolberger** [comment]
@artpumpkin The mentioned props look like they are for `framer-motion` components. Please see #2538 for a solution.

**Comment by mike-fam** [comment]
@TimKolberger I'm not sure how to apply the fix at #2538 to my problem. Can you show me how to fix the CodeSandbox example above?

**Comment by TimKolberger** [comment]
@mike-fam my previous comment was directed to artpumpkin.

The issue with isFullWidth needs a fix in our codebase. We need to exclude isFullWidth from spreading to the DOM element.
